[Royce Ho Shou Yee] iP

TheSpaceCuber opened the pull request:
Added TODO list features.

Commit:
Add Gradle support

TheSpaceCuber committed:
Level-1

TheSpaceCuber committed:
Level-2

TheSpaceCuber committed:
level-3

TheSpaceCuber committed:
Level-4

TheSpaceCuber committed:
Level-6

TheSpaceCuber committed:
A-Enums

TheSpaceCuber committed:
fixed exceptions

TheSpaceCuber committed:
update exceptions and tags

TheSpaceCuber committed:
Level-7

TheSpaceCuber committed:
Level-7

TheSpaceCuber committed:
Level-8

TheSpaceCuber committed:
Merge branch 'branch-Level-8'
updating master to Level 8: Datetime implementation

TheSpaceCuber committed:
adding data.txt to commit

TheSpaceCuber committed:
A-MoreOOP

TheSpaceCuber committed:
A-Packages

TheSpaceCuber committed:
A-JUnit

TheSpaceCuber committed:
A-JavaDoc added

TheSpaceCuber committed:
test cases

TheSpaceCuber committed:
Level-9 added

TheSpaceCuber committed:
Merge branch 'add-gradle-support'

Merge gradle branch into master for Gradle setup

TheSpaceCuber committed:
Completed A-Gradle and A-Checkstyle. Incorporated
Grade for automation and fixed checkstyle issues.

TheSpaceCuber committed:
removed gradle files as there are issues with gradle

TheSpaceCuber committed:
patched gradle issues by reinstalling gradle to project

TheSpaceCuber committed:
patched gradle issues by reinstalling gradle to project

wilinetan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,109 @@
+            ui.showLine();
+            try {
+                Choice c = parser.parseCommand(input);
+                switch (c) {
```
I like how you used a switch statement to check for the type of command

wilinetan commented on src/main/java/utility/Choice.java:
```diff
@@ -0,0 +1,5 @@
+package utility;
+
+public enum Choice {
```
I like how you used an enum class for the different commands

maxxyh replied:
Agreed! It's clean!

wilinetan commented on src/main/java/utility/Storage.java:
```diff
@@ -0,0 +1,118 @@
+                newList.add(new Todo(line.substring(7)));
+                break;
+            case 'E':
+                int posE = getPosition(line, '/');
```
Perhaps the variable name can just be pos?

wilinetan commented on src/main/java/utility/Storage.java:
```diff
@@ -0,0 +1,118 @@
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+
+
+
+
+
+
+
+
+
+
+
```
The empty spaces might not be required?

TheSpaceCuber replied:
thanks!

wilinetan reviewed (commented):
Overall, good documenting and consistent effort in following the coding standards

TheSpaceCuber committed:
Implemented GUI using JavaFX, application now has a basic interface

TheSpaceCuber committed:
added images for user and duke in UI

TheSpaceCuber committed:
Added assertions to tasklist operations such as addDeadline etc

TheSpaceCuber committed:
fixed code quality issues. edited to follow coding standards.

TheSpaceCuber committed:
Merge pull request #1 from TheSpaceCuber/branch-A-CodeQuality

fixed code quality issues. edited to follow coding standards.

maxxyh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,86 @@
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Nitpicking here but this javadocs you can replace with your own!

maxxyh commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+     *
+     * @return Text format to be saved.
+     */
+    public String toWrite() {
```
Perhaps a name like getSaveString() would be more intuitive?

maxxyh commented on src/main/java/utility/Parser.java:
```diff
@@ -0,0 +1,32 @@
+     * @throws DukeException Duke Custom Exception.
+     */
+    public Choice parseCommand(String input) throws DukeException {
+        ArrayList<String> commands = new ArrayList<>(
+                Arrays.asList("DONE", "LIST", "DEADLINE", "EVENT", "TODO", "BYE", "DELETE", "FIND"));
```
Perhaps these values could be obtained directly from the Choice Enum class like [here](https://stackoverflow.com/questions/13783295/getting-all-names-in-an-enum-as-a-string)! That would mean less updating 👍

maxxyh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,86 @@
+        try {
+            Choice c = parser.parseCommand(input);
+            System.out.println(c);
+            switch (c) {
```
Perhaps instead of using the full input for each task, you could consider using the parser to prepare the input accordingly? Cos I noticed you work with substrings within the individual task methods, causing the taskList to be quite lengthy! So yeah perhaps some of that data processing responsibility can be abstracted to the parser?

maxxyh commented on src/main/java/utility/Storage.java:
```diff
@@ -0,0 +1,108 @@
+     * @return ArrayList that will be the TaskList.
+     * @throws FileNotFoundException exception if file isn't found.
+     */
+    public ArrayList<Task> load() throws FileNotFoundException {
```
I'm not entirely sure but I think IOException might be a possibility here too!

maxxyh commented on src/main/java/utility/Ui.java:
```diff
@@ -0,0 +1,51 @@
+     * Prints out the welcome text.
+     */
+    public void showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        String msg = "____________________________________________________________\n"
+                + "Hello I'm Duke\n"
+                + "Your personal smart assistant\n"
+                + "____________________________________________________________";
```
Perhaps these strings could be abstracted as constant variables at the top of the class? Would be easier to edit and cleaner too!

maxxyh commented on src/test/java/utility/ParserTest.java:
```diff
@@ -0,0 +1,26 @@
+        assertEquals(2, 2);
+    }
+
+    @Test
+    public void parseCommandTest() throws DukeException {
+        Parser parser = new Parser();
+        Choice actualOutput = parser.parseCommand("bye");
+        assertEquals(Choice.BYE, actualOutput);
+    }
+
+    @Test
+    public void parseCommandTest2() throws DukeException {
+        Parser parser = new Parser();
+        Choice actualOutput = parser.parseCommand("done 2");
+        assertEquals(Choice.DONE, actualOutput);
```
I like that your code is decoupled and easy to test!

maxxyh reviewed (commented):
LGTM, overall very clean code and great javadocs! Maybe can consider using the Command pattern just as a form of personal practice!

TheSpaceCuber committed:
fixed unicode bug in displaying ticks and axes in Task.java

TheSpaceCuber committed:
added an extension to delete multiple tasks at once. For e.g. delete 1 2 will delete task 1 and 2 instantly.

TheSpaceCuber committed:
Update README to latest requirements.

Included references to tutorials as well as
fine-tune to fit gradle, junit etc.

TheSpaceCuber committed:
Add Ui.png file to docs folder

TheSpaceCuber committed:
Add README to docs folder as well.

TheSpaceCuber committed:
Restore User Guide to default template

TheSpaceCuber committed:
Merge pull request #2 from TheSpaceCuber/branch-Level-7

Restore User Guide to default template

TheSpaceCuber committed:
Minor edit to UG

TheSpaceCuber committed:
Merge pull request #3 from TheSpaceCuber/branch-Level-8

Minor edit to UG

TheSpaceCuber committed:
New edit to UG

TheSpaceCuber committed:
Merge pull request #4 from TheSpaceCuber/branch-A-JavaDoc

New edit to UG

TheSpaceCuber committed:
Third feature add to UG

TheSpaceCuber committed:
Merge pull request #5 from TheSpaceCuber/branch-A-CodingStandard

Third feature add to UG

TheSpaceCuber committed:
Add signal to peer reviewer

TheSpaceCuber committed:
Merge pull request #6 from TheSpaceCuber/branch-Level-9

Add signal to peer reviewer

TheSpaceCuber committed:
Dummy commit for branch tagging

TheSpaceCuber committed:
Merge pull request #7 from TheSpaceCuber/branch-Level-10

Dummy commit for branch tagging

TheSpaceCuber committed:
Dummy commit for branch tagging

TheSpaceCuber committed:
Merge pull request #8 from TheSpaceCuber/branch-A-Assertions

Dummy commit for branch tagging

TheSpaceCuber committed:
Update User Guide and include photos for UG

TheSpaceCuber committed:
Add photos for UG

TheSpaceCuber committed:
Rename certain titles in UG

TheSpaceCuber committed:
Naming and filename fix

TheSpaceCuber committed:
Set theme jekyll-theme-cayman
